server/views/tasks/_navigation.njk

```diff
@@ -1,19 +1,19 @@
 {%- from "moj/components/sub-navigation/macro.njk" import mojSubNavigation -%}
 
-{% macro navigation(activeTab) %}
+{% macro navigation(activeTab, apArea) %}
   {{
     mojSubNavigation({
       label: 'Sub navigation',
       items: [
         {
           text: 'Allocated',
-          href: paths.tasks.index({})+ "?allocatedFilter=allocated",
-          active: (activeTab === 'allocated' or activeTab|length === 0)
+          active: (activeTab === 'allocated' or activeTab|length === 0),
+          href: (paths.tasks.index({})+ "?allocatedFilter=allocated&area="+apArea if apArea else paths.tasks.index({})+ "?allocatedFilter=allocated")
         },
         {
           text: 'Unallocated',
-          href: paths.tasks.index({}) + "?allocatedFilter=unallocated",
-          active: (activeTab === 'unallocated')
+          active: (activeTab === 'unallocated'),
+          href: (paths.tasks.index({})+ "?allocatedFilter=unallocated&area="+apArea if apArea else paths.tasks.index({})+ "?allocatedFilter=unallocated")
         }
       ]
```

Comment:
This is getting a little complicated - is there a suitable utils file this could be done in instead? It might be worth moving the whole `items` array generation into a utils method and then pass that to the template from the controller